### PR TITLE
Fixed #27802 -- Unified datetime sql functions with others backend operations behavior.

### PR DESCRIPTION
**Pull request by felixxm**
Ticket [27802](https://code.djangoproject.com/ticket/27802).

### CODE REVIEW

**Comment by charettes** [comment]
It might be worth a note in the `Database backends` section of the 2.0 release notes.

**Comment by claudep** [comment]
And we have to be absolutely sure timezone names cannot be hacked, as this would allow SQL injection.

**Comment by felixxm** [comment]
@charettes, @claudep Thanks for review. I added release notes.

We use these functions only in [`db/models/functions/datetime.py`](https://github.com/django/django/blob/master/django/db/models/functions/datetime.py) and all of them get timezone name from [`timezone._get_timezone_name()`](https://github.com/django/django/blob/master/django/utils/timezone.py#L107).

File: docs/releases/2.0.txt [hunk 1]
```diff
@@ -208,7 +208,11 @@ Backwards incompatible changes in 2.0
 Database backend API
 --------------------
 
-* ...
+* The ``DatabaseOperations.datetime_cast_date_sql()``,
+  ``DatabaseOperations.datetime_cast_time_sql()``,
+  ``DatabaseOperations.datetime_trunc_sql()``,
+  ``DatabaseOperations.datetime_extract_sql()`` methods are now return the SQL
```
Comment (jarshwah) [inline docs/releases/2.0.txt]: The "are" isn't necessary here

Comment (felixxm) [inline docs/releases/2.0.txt]: @jarshwah Thanks for review. I updated release notes.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by timgraham** [comment]
merged in 15c14f6f16af22b970e5693b349113d80f10f82d, thanks!

**Pull request closed without merging**